### PR TITLE
Add pagination for direct-helm get summaries.

### PR DESCRIPTION
**Pull request by absoludity**
### Description of the change

Adds and tests pagination for direct helm plugin's GetAvailablePackageSummaries request.

### Applicable issues

  - Ref #2993

### CODE REVIEW

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server_test.go [hunk 1]
```diff
@@ -576,29 +611,135 @@ func TestGetAvailablePackageSummaries(t *testing.T) {
...
+
 			if tc.expectDBQuery {
 				// Checking if the WHERE condtion is properly applied
 				mock.ExpectQuery("SELECT info FROM").
 					WithArgs(tc.request.Context.Namespace, tc.server.globalPackagingNamespace).
 					WillReturnRows(rows)
+				if tc.request.GetPaginationOptions().GetPageSize() > 0 {
+					mock.ExpectQuery("SELECT count").
+						WithArgs(tc.request.Context.Namespace, tc.server.globalPackagingNamespace).
+						WillReturnRows(sqlmock.NewRows([]string{"count"}).AddRow(3))
```
Comment (antgamdia) [inline cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server_test.go:741]: Is it always `3` ?

**Review by antgamdia — APPROVED** [review]
Ooops, I forgot to submit the review yesterday :facepalm: Sorry!
Thanks for the changes here! Happy to finally see the pagination!

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go [hunk 2]
```diff
@@ -149,8 +150,12 @@ func (s *Server) GetAvailablePackageSummaries(ctx context.Context, request *core
 		cq.AppVersion = request.FilterOptions.AppVersion
 	}
 
-	// TODO: We are not yet returning paginated results here
-	charts, _, err := s.manager.GetPaginatedChartListWithFilters(cq, 1, 0)
+	pageSize := request.GetPaginationOptions().GetPageSize()
```
[the remark this answers is not in the document]
Comment (absoludity) [reply cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go:153]: Yeah, though it's specific to the grpc message generation, which provides these getters that initialise to the default (empty) struct.

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go [hunk 3]
```diff
@@ -164,11 +169,37 @@ func (s *Server) GetAvailablePackageSummaries(ctx context.Context, request *core
...
+	}
 	return &corev1.GetAvailablePackageSummariesResponse{
 		AvailablePackagesSummaries: responsePackages,
+		NextPageToken:              nextPageToken,
 	}, nil
 }
 
+// pageOffsetFromPageToken converts a page token to an integer offset
+// representing the page of results.
+// TODO(mnelson): When aggregating results from different plugins, we'll
```
[the remark this answers is not in the document]
Comment (absoludity) [reply cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go:187]: Yep. And we can leave it that way until we're updating to built-in cache support for plugins, I'd think.

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server_test.go (hunk 1 above)
Comment (absoludity) [reply cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server_test.go:741]: Yeah, it's a query in the assetsvc that returns the total pages (so counts the filtered but not paginated set): https://github.com/kubeapps/kubeapps/blob/master/cmd/assetsvc/pkg/utils/postgresql_utils.go#L80

**Review by absoludity — COMMENTED** [review]
> Ooops, I forgot to submit the review yesterday facepalm Sorry!

No worries - done the same many times :) Thanks!

**Pull request merged**